pkg/metaserver/agent/metric/malachite/fetcher_calculate.go

```diff
@@ -27,45 +27,48 @@ import (
 )
 
 // processContainerMemBandwidth handles memory bandwidth (read/write) rate in a period while,
-// and it will need the previously collected datat to do this
-func (m *MalachiteMetricsFetcher) processContainerMemBandwidth(podUID, containerName string, cgStats *types.MalachiteCgroupInfo) {
+// and it will need the previously collected data to do this
+func (m *MalachiteMetricsFetcher) processContainerMemBandwidth(podUID, containerName string, cgStats *types.MalachiteCgroupInfo, lastUpdateTimeInSec float64) {
 	var (
 		lastOCRReadDRAMs, _ = m.metricStore.GetContainerMetric(podUID, containerName, consts.MetricOCRReadDRAMsContainer)
 		lastIMCWrites, _    = m.metricStore.GetContainerMetric(podUID, containerName, consts.MetricIMCWriteContainer)
 		lastStoreAllIns, _  = m.metricStore.GetContainerMetric(podUID, containerName, consts.MetricStoreAllInsContainer)
 		lastStoreIns, _     = m.metricStore.GetContainerMetric(podUID, containerName, consts.MetricStoreInsContainer)
-		lastUpdateTime, _   = m.metricStore.GetContainerMetric(podUID, containerName, consts.MetricUpdateTimeContainer)
 	)
 
 	var readBandwidth, writeBandwidth float64
-	var curOCRReadDRAMs, curIMCWrites, curStoreAllIns, curStoreIns, curUpdateTime float64
+	var curOCRReadDRAMs, curIMCWrites, curStoreAllIns, curStoreIns, curUpdateTimeInSec float64
 	if cgStats.CgroupType == "V1" {
 		curOCRReadDRAMs = float64(cgStats.V1.Cpu.OCRReadDRAMs)
 		curIMCWrites = float64(cgStats.V1.Cpu.IMCWrites)
 		curStoreAllIns = float64(cgStats.V1.Cpu.StoreAllInstructions)
 		curStoreIns = float64(cgStats.V1.Cpu.StoreInstructions)
-		curUpdateTime = float64(cgStats.V1.Cpu.UpdateTime)
+		curUpdateTimeInSec = float64(cgStats.V1.Cpu.UpdateTime)
 	} else if cgStats.CgroupType == "V2" {
 		curOCRReadDRAMs = float64(cgStats.V2.Cpu.OCRReadDRAMs)
 		curIMCWrites = float64(cgStats.V2.Cpu.IMCWrites)
 		curStoreAllIns = float64(cgStats.V2.Cpu.StoreAllInstructions)
 		curStoreIns = float64(cgStats.V2.Cpu.StoreInstructions)
-		curUpdateTime = float64(cgStats.V2.Cpu.UpdateTime)
+		curUpdateTimeInSec = float64(cgStats.V2.Cpu.UpdateTime)
 	}
 
 	// read/write bandwidth calculation formula
-	timediffSecs := curUpdateTime - lastUpdateTime.Value
-	if timediffSecs > 0 {
-		readBytes := (curOCRReadDRAMs - lastOCRReadDRAMs.Value) * 64
-		readBandwidth = readBytes / (1024 * 1024 * timediffSecs)
+	timeDiffSecs := curUpdateTimeInSec - lastUpdateTimeInSec
+	if timeDiffSecs <= 0 {
```

Comment:
@justadogistaken so this pr only fixes the case that when `timeDiffSecs` <= 0, we should skip the calculation  rather than set the values as zero by mistake. right?

Reply:
Yes. I found that it produced lots of "zero" value. The problem is originated from the sampling gap between katalyst(once per 5s) and malachite-core(once per 10s).

And I'm checking if there are some other metrics have the same problem, I will fix them along this pr.

Reply:
btw, the coding style is really elegant!